File: internal/domeneshop/dyndns.go
package domeneshop

import (
	"context"
	"net/url"
)

// UpdateDynDNS updates the DNS record for hostname to point at myip (or the
// caller's apparent IP address if myip is empty), using Domeneshop's
// dynamic DNS update protocol.
func (c *Client) UpdateDynDNS(ctx context.Context, hostname, myip string) error {
	query := url.Values{"hostname": {hostname}}
	if myip != "" {
		query.Set("myip", myip)
	}
	_, err := c.request(ctx, "GET", "/dyndns/update", query, nil, nil)
	return err
}

File: src/domeneshop/resources/dyndns.ts
import type { HttpTransport } from "../http.js";

export class DynDnsResource {
  constructor(private readonly transport: HttpTransport) {}

  /** Omitting myIp resolves it to the caller's apparent IP address instead. */
  async update(hostname: string, myIp?: string): Promise<void> {
    await this.transport.request("GET", "/dyndns/update", {
      query: { hostname, myip: myIp },
      decodeJson: false,
    });
  }
}
